pkg/kv/kvserver/rangefeed/buffered_registration.go

```diff
@@ -120,6 +120,10 @@ func (br *bufferedRegistration) publish(
 	if br.mu.overflowed {
```

Comment:
drive-by: is it intentional that publish holds an exclusive mutex below even in the test case in which we do a blocking send?

Reply:
Hmm, I'm not entirely sure if I follow. I think we need to hold an exclusive mutex here since we write to this variable here https://github.com/cockroachdb/cockroach/blob/5983b096046893b1a95dcfe9fbf1150d4fb95201/pkg/kv/kvserver/rangefeed/buffered_registration.go#L146 and then read it in another goroutine, `runOutputLoop` https://github.com/cockroachdb/cockroach/blob/5983b096046893b1a95dcfe9fbf1150d4fb95201/pkg/kv/kvserver/rangefeed/buffered_registration.go#L216. Perhaps Steven might have some insights.


Which test case are you referring to? Are you talking about the one where we cause `stream.Send` to be blocked https://github.com/cockroachdb/cockroach/blob/c48acff77f042cc57470031db7191594f1621f64/pkg/kv/kvserver/rangefeed/unbuffered_sender_test.go#L167? Or are you referring more generally to a send to the gRPC stream without buffering events?
<!-- Reviewable comment -OC9PkLEAXaW34cNazyV:b8nu196 -->
<!-- Sent from Reviewable.io -->